openstack/README.md

```diff
@@ -0,0 +1,103 @@
+  - Make sure you have the required flavors on OpenStack by enabling the [flavors extension](https://github.com/cloudfoundry-incubator/cf-openstack-validator/tree/master/extensions/flavors) with the [`flavors.yml`](./flavors.yml) file in this directory. Flavor names need to match those specified in the cloud config.
+  - If you plan using the [Swift ops file](use-swift-blobstore.yml) to enable Swift as blobstore for the Cloud Controller, you should also run the [Swift extension](https://github.com/cloudfoundry-incubator/cf-openstack-validator/tree/master/extensions/object_storage).
+
+## Prepare OpenStack resources for BOSH and Cloud Foundry via Terraform
+
+### BOSH
+
+To setup an OpenStack project to install BOSH please use the following [Terraform module](https://github.com/cloudfoundry-incubator/bosh-openstack-environment-templates/tree/master/bosh-init-tf). Adapt `terraform.tfvars.template` to your needs.
+
+### Cloud Foundry
+
+To setup the project to install Cloud Foundry please use the following [Terraform module](https://github.com/cloudfoundry-incubator/bosh-openstack-environment-templates/tree/master/cf-deployment-tf). Adapt `terraform.tfvars.template` to your needs. Variable `bosh_router_id` is output of the previous BOSH terraform module.
+
+## Install BOSH
+
+To install the BOSH director please follow the instructions on [bosh.io](https://bosh.io/docs/init-openstack.html#deploy).
+
+Make sure the BOSH director is accessible through the BOSH cli, by following the instructions on [bosh.io](https://bosh.io/docs/cli-envs.html). Use this mechanism in all BOSH cli examples in
+this documentation.
+
+## Cloud Config
+
+After the BOSH director has been installed, you can prepare and upload a cloud config based on the [cloud-config.yml](../cloud-configs/openstack/cloud-config.yml) file.
+
+Take the variables and outputs from the Terraform run of `cf-deployment-tf` to finalize the cloud config.
+
+Use the following command to upload the cloud config.
+```
+bosh update-cloud-config \
+  -v availability_zone1="<az-1>" \
+  -v availability_zone2="<az-2>" \
+  -v availability_zone3="<az-3>" \
+  -v network_id1="<cf-network-id-1>" \
+  -v network_id2="<cf-network-id-2>" \
+  -v network_id3="<cf-network-id-3>" \
+  ../cloud-configs/openstack/cloud-config.yml
+```
+
+## Deploy Cloud Foundry
+
+To deploy Cloud Foundry run the following command filling in the necessary variables. `system_domain` is the user facing domain name of your Cloud Foundry installation.
+
+```
+bosh -d cf deploy ../cf-deployment.yml \
+  -o ../operations/openstack/use-compiled-releases.yml \
+  -o ../operations/openstack/ephemeral-disk-vm-extensions.yml \
+  --vars-store ~/deployments/cf/cf-vars-store.yml \
+  -v system_domain="<system-domain>"
+```
+
+### With TCP Routing
+
+If you want to enable TCP Routing, the Terraform module should have run with `use_tcp_routing=true`. Add the following line to the deploy cmd:
```

Comment:
@voelzmo I think you might have missed this, but we recently inlined TCP Routing from the ops-file into cf-deployment.yml. I think we can just get rid of this section  -- assuming the terraform template creates TCP LBs by default, which seems to be the case.

Reply:
Nice, we did miss that, thanks!